nifi-commons/nifi-site-to-site-client/src/main/java/org/apache/nifi/remote/codec/StandardFlowFileCodec.java

```diff
@@ -56,13 +62,32 @@ public void encode(final DataPacket dataPacket, final OutputStream encodedOut) t
             writeString(entry.getValue(), out);
         }
 
-        out.writeLong(dataPacket.getSize());
+        final long dataSize = dataPacket.getSize();
+        out.writeLong(dataSize);
 
         final InputStream in = dataPacket.getData();
-        StreamUtils.copy(in, encodedOut);
+        copy(dataSize, in, encodedOut);
         encodedOut.flush();
     }
 
+    private long copy(long length, final InputStream source, final OutputStream destination) throws IOException {
+
+        final StopWatch stopWatch = new StopWatch(false);
+        final byte[] buffer = new byte[8192];
+        long totalRead = 0;
+        // Use 'available' so that it doesn't block
+        for (int available; totalRead < length && (available = source.available()) > 0;) {
```

Comment:
We cannot use InputStream.available() for this purpose. See https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#available() for an explanation of the method. It is an "estimate" of the number of bytes, but a great number of InputStreams simply return `0`, which makes the result unreliable. In such a case, this will immediately return without copying any bytes.

Reply:
Thank you for catching this. I think this is a leftover while I was trying different things. Reverted changes to StandardFlowFileCodec.